### PR TITLE
Implement i128/u128 to JsBigInt conversions

### PR DESCRIPTION
**Pull request by AlvinKuruvilla**
This commit implements trait conversions to JsValue for i128/u128. This addresses #1970

This Pull Request fixes/closes #{1970}.

It changes the following:

- Adds trait conversions for the missing i128 and u128 rust types to JsValue
-
-

### CODE REVIEW

**Comment by AlvinKuruvilla** [comment]
Yeah that makes sense, I saw this [issue](https://github.com/boa-dev/boa/issues/1970) open and a lack of activity, so I thought it might be a good way to get my feet wet a little bit. Is there no longer any interest in doing this though?

**Comment by jedel1043** [comment]
> Hmmm I'm not sure if we want to do this. There is no native type in JavaScript equivalent to i/u128. And using a BigInt is not the same thing, since it's stored in the heap and performance is different.

Note that this PR implements `From` for `JsBigInt` and `Numeric`, not `JsValue`, and both traits are also implemented for `i/u64`.

**Comment by jedel1043** [comment]
I'd support merging this change. Converting to `Numeric` is a lot more explicit about allocations than directly converting to `JsValue`, and if we want to make it a lot more explicit we could specify on the `Numeric` docs that all conversions from integer types allocate.

File: boa_engine/src/bigint.rs [hunk 1]
```diff
@@ -407,7 +407,22 @@ impl From<usize> for JsBigInt {
...
+    }
+}
+impl From<u128> for JsBigInt {
+    #[inline]
+    fn from(value: u128) -> Self {
+        Self {
+            inner: Rc::new(RawBigInt::from(value)),
+        }
+    }
+}
```
Comment (HalidOdat) [inline boa_engine/src/bigint.rs:425]: This is fine (conversion from `u/i128` to `JsBigInt`), but I don't think we should add for `Numeric`.

Comment (AlvinKuruvilla) [reply boa_engine/src/bigint.rs:425]: Okay I can make that change

Comment (AlvinKuruvilla) [reply boa_engine/src/bigint.rs]: Resolved by the latest commit to the PR, thanks for the feedback @HalidOdat

Comment (HalidOdat) [reply boa_engine/src/bigint.rs]: I meant that we should keep the `u/i128` to `JsBigInt` conversion, and only remove the `u/i128` to `Numeric` conversion.

Comment (AlvinKuruvilla) [reply boa_engine/src/bigint.rs]: Oh sorry. But the `JsBigInt` conversion requires `Numeric` trait to be implemented

Comment (AlvinKuruvilla) [reply boa_engine/src/bigint.rs]: Sorry, that was my mistake I had it backward, I re-added the trait impl for `JsBigInt`

File: boa_engine/src/bigint.rs [hunk 2]
```diff
@@ -407,7 +422,6 @@ impl From<usize> for JsBigInt {
         }
     }
 }
-
```
Comment (HalidOdat) [inline boa_engine/src/bigint.rs:410]: Could we add the newline back here.

File: boa_engine/src/bigint.rs [hunk 3]
```diff
@@ -389,7 +389,22 @@ impl From<u64> for JsBigInt {
...
+impl From<i128> for JsBigInt {
+    #[inline]
+    fn from(value: i128) -> Self {
+        Self {
+            inner: Rc::new(RawBigInt::from(value)),
+        }
+    }
+}
+impl From<u128> for JsBigInt {
```
Comment (HalidOdat) [inline boa_engine/src/bigint.rs:400]:
```suggestion
}

impl From<u128> for JsBigInt {
```

Add new line like other functions.

**Review by HalidOdat — APPROVED** [review]
Looks good! just some small nitpicks :)

**Review by jedel1043 — APPROVED** [review]

**Pull request merged**